[docs] batch 12: reduction functions (#16930)
apache/druid: docs/querying/sql-functions.md

## GREATEST

> Returns the maximum value from the provided expressions. The expressions must all be convertible to a common data type.

Reviewer: What does "convertible to a common data type" mean? Convertible by whom? By Druid? Data type common to Druid? I think this sentence could use some clarification.

> Returns the maximum value from the provided expressions. The expressions must all be convertible to a common data type. 

Reviewer 2: ```suggestion
Returns the maximum value from the provided expressions. The expressions must all be convertible to a common data type.
```

> Returns the maximum value from the provided expressions. The [following rules](sql-scalar.md#reduction-functions) determine how Druid compares between different data types.

Reviewer: ```suggestion
Returns the maximum value from the provided expressions. For information on how Druid interprets the arguments passed into the function, see [Reduction functions](sql-scalar.md#reduction-functions).
```
Since these aren't exactly rules, I think it's best to rephrase.

* **Syntax:** `GREATEST([expr1, ...])`
* **Function type:** Scalar, reduction

<details><summary>Example</summary>

> The following example returns the greatest value between the numeric constant `PI`, the integer number `4` and the double `-5.0`, which are all interpreted as doubles. 

Reviewer: ```suggestion
The following example returns the greatest value between the numeric constant `PI`, the integer number `4`, and the double `-5.0`, which are all interpreted as doubles. 
```
This sentence is too long. I suggest that you split it into two. Also, "interpreted as doubles" by whom? Clarify and, if possible, use active voice.

> The following example returns the greatest value between the numeric constant `PI`, the integer number `4`, and the double `-5.0`. Druid interprets all expressions as doubles to make comparison between integers and doubles possible. 

Reviewer: ```suggestion
The following example returns the greatest value between the numeric constant `PI`, the integer number `4`, and the double `-5.0`. Druid interprets all expressions as DOUBLE to make comparison between integers and doubles possible. 
```
Changed doubles to DOUBLE to match the style in https://druid.apache.org/docs/latest/querying/sql-data-types#standard-types

## LEAST

> Returns the minimum value from the provided arguments. The expressions must all be convertible to a common data type. 

Reviewer: Should this say "expressions" instead of "arguments" for consistency? Also, this [this comment](https://github.com/apache/druid/pull/16930/files#r1742738370) regarding "convertible to a common data type."

> * **Syntax** `LEAST([expr1, ...])`

Reviewer 2: ```suggestion
* **Syntax:** `LEAST([expr1, ...])`
```

* **Function type:** Scalar, reduction

<details><summary>Example</summary>

> The following example returns the least value between the strings `apple`, `orange` and `pear` , which already share a common datatype.

Reviewer 2: ```suggestion
The following example returns the least value between the strings `apple`, `orange` and `pear`, which already share a common datatype.
```

> The following example returns the least value between the strings `apple`, `orange` and `pear`, which already share a common datatype.

Reviewer: "least values" sounds a bit odd. Consider "the minimum value" or " the smallest value".

> The following example returns the least value between the strings `apple`, `orange` and `pear`, which already share a common datatype.

Reviewer: ```suggestion
The following example returns the least value between the strings `apple`, `orange`, and `pear`, which already share a common datatype.
```
I suggest that you split this sentence into two to improve readability.

> The following example returns the minimum value between the strings `apple`, `orange`, and `pear`. Druid compares the expressions as strings because they're all already in string format. 

Reviewer: ```suggestion
The following example returns the minimum value between the strings `apple`, `orange`, and `pear`. Druid interprets these arguments as STRING data type. 
```
